crates/bevy_math/src/primitives/dim3.rs

```diff
@@ -796,8 +796,9 @@ impl Bounded3d for Triangle3d {
     fn bounding_sphere(&self, translation: Vec3, rotation: Quat) -> BoundingSphere {
         if self.is_degenerate() {
             let (p1, p2) = self.largest_side();
+            let mid_point = (p1 + p2) / 2.0;
             let (segment, _) = Segment3d::from_points(p1, p2);
-            return segment.bounding_sphere(translation, rotation);
+            return segment.bounding_sphere(translation + mid_point, rotation);
```

Comment:
Is using Segment3d doing us any favours here? I'd wager that we can write out the whole computation here, as we already do most of it.

Reply:
Another idea:
The code could be re-done to work like:
```rust
if self.is_degenerate() || self.is_obtuse() {
    let (p1, p2) = self.largest_side();
    let mid_point = (p1 + p2) / 2.0;
    let (segment, _) = Segment3d::from_points(p1, p2);
    return segment.bounding_sphere(translation + mid_point, rotation);
}
```
Because `side opposite to obtuse` =(assuming triangle is obtuse)= `side opposite to largest angle` = `largest side`. This would let us skip the whole other code-path which determines this.

Another reason this is relevant, given the original line 
```rust
            let (segment, _) = Segment3d::from_points(p1, p2);
            return segment.bounding_sphere(translation, rotation);
```
is incorrect, then I believe there is another same mistake at line 817-818.

Merging these two code-paths would make this fix both cases.
Alternatively we should at least fix the other code path without merging them yet.